apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.spec.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @jest-environment node
+ */
+
+const dayjs = require('dayjs')
+
+const { setFakeClientMode } = require('@core/keystone/test.utils')
+const { sendSubmitMeterReadingsPushNotifications } = require('@condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications')
+const { Message: MessageApi } = require('@condo/domains/notification/utils/serverSchema')
+const {
+    METER_SUBMIT_READINGS_REMINDER_TYPE,
+    METER_VERIFICATION_DATE_EXPIRED_TYPE,
+} = require('@condo/domains/notification/constants/constants')
+const {
+    MeterReadingSource,
+    makeClientWithResidentAndMeter,
+    createTestMeterReading,
+} = require('@condo/domains/meter/utils/testSchema')
+const { CALL_METER_READING_SOURCE_ID } = require('@condo/domains/meter/constants/constants')
+
+const index = require('@app/condo/index')
+const { keystone } = index
+
+const taskRunTimeoutMs = 30000
+
+const prepareUserAndMeter = async ({ nextVerificationDate }) => {
+    return await makeClientWithResidentAndMeter({
+        verificationDate: dayjs().add(-1, 'year').toISOString(),
+        nextVerificationDate,
+    })
+}
+
+const prepareReadings = async (client) => {
+    const [source] = await MeterReadingSource.getAll(client, { id: CALL_METER_READING_SOURCE_ID })
+
+    await createTestMeterReading(
+        client,
+        client.meter,
+        client.organization,
+        source,
+        {
+            date: dayjs().toISOString(),
+        }
+    )
+}
+
+const getNewMessages = async ({ userId, meterId }) => {
+    const messages = await MessageApi.getAll(keystone, {
+        user: { id: userId },
+        type_in: [METER_SUBMIT_READINGS_REMINDER_TYPE, METER_VERIFICATION_DATE_EXPIRED_TYPE],
+    })
+    return messages.filter(message => message.meta.data.meterId === meterId)
+}
+
+describe('Submit meter readings push notification', () => {
```

Comment:
You don't have some important age cases:

 1) it should work with `nextVerificationDate = undefined`

 2) your code should not fail in case of sendMessage key duplication! It raise an exception!

Reply:
it doesn't fail in case of key duplication. There are log of exception at sendMessage level (I can't handle that). In my code I'm handling that in sendMessageSafely function level